src/sonnenhut/common.py

```diff
@@ -61,6 +61,32 @@ def goldenhour(location, direction):
     return location.golden_hour(direction=direction, date=None, local=True)
 
 
+def safeget(dct, *keys, default='?'):
+    """Helper function to hide the ugliness of accessing nested dictionaries with lists
+
+    >>> exampledict = {'a': {'b': 1, 'c': [{'aa': 23, 'bb': 45}], 'd': 3}}
+    >>> safeget(exampledict, 'a', 'b')
+    1
+    >>> safeget(exampledict, 'a', 'x')
+    '?'
+    >>> safeget(exampledict, 'a', 'x', default='???')
+    '???'
+    >>> safeget(exampledict, 'a', 'c', 0, 'bb')
+    45
+
+    :param dict dct: Dictionary
+    :param tuple keys: Keys to access, usually strings or integers
+    :param str default: the default value if the keys couldn't be accessed
+    :return: the value in the dictionary or the default value
+    """
+    for key in keys:
+        try:
+            dct = dct[key]
+        except KeyError:
```

Comment:
Yes, this will only catch `KeyError` exceptions. Not entirely sure what you mean by *out of range error*. I guess you mean `IndexError`?

This should be added to the except line, yes. ☺